app/controlplane/pkg/auditor/events/workflowcontract.go

```diff
@@ -0,0 +1,183 @@
+//
+// Copyright 2024 The Chainloop Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package events
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	"github.com/chainloop-dev/chainloop/app/controlplane/pkg/auditor"
+
+	"github.com/google/uuid"
+)
+
+var (
+	_ auditor.LogEntry = (*WorkflowContractCreated)(nil)
+	_ auditor.LogEntry = (*WorkflowContractUpdated)(nil)
+	_ auditor.LogEntry = (*WorkflowContractDeleted)(nil)
+	_ auditor.LogEntry = (*WorkflowContractAttached)(nil)
+	_ auditor.LogEntry = (*WorkflowContractDetached)(nil)
+)
+
+const (
+	WorkflowContractType                       auditor.TargetType = "WorkflowContract"
+	WorkflowContractCreatedActionType          string             = "WorkflowContractCreated"
+	WorkflowContractUpdatedActionType          string             = "WorkflowContractUpdated"
+	WorkflowContractDeletedActionType          string             = "WorkflowContractDeleted"
+	WorkflowContractContractAttachedActionType string             = "WorkflowContractContractAttached"
+	WorkflowContractContractDetachedActionType string             = "WorkflowContractContractDetached"
+)
+
+// WorkflowContractBase is the base struct for workflow contract events
+type WorkflowContractBase struct {
```

Comment:
Thanks @javirln!

BTW, I am curious to see what's your take on the testing framework I put here for the generation of events, let me know what you think.